src/Simulation/Simulators/SparseSimulator/SparseSimulatorCS/Probes.cs

```diff
@@ -0,0 +1,360 @@
+            }
+            else if (angle < 0)
+            {
+                var abs_angle = System.Math.Abs(angle);
+                if (abs_angle >= (0 * PI / 8) + offset && abs_angle < ((1 * PI / 8) + offset)) { chart = "     \\+"; }
+                if (abs_angle >= (1 * PI / 8) + offset && abs_angle < ((2 * PI / 8) + offset)) { chart = "     \\ "; }
+                if (abs_angle >= (2 * PI / 8) + offset && abs_angle < ((3 * PI / 8) + offset)) { chart = "    -\\ "; }
+                if (abs_angle >= (3 * PI / 8) + offset && abs_angle < ((4 * PI / 8) + offset)) { chart = "   ↓   "; }
+                if (abs_angle >= (4 * PI / 8) + offset && abs_angle < ((5 * PI / 8) + offset)) { chart = " /+    "; }
+                if (abs_angle >= (5 * PI / 8) + offset && abs_angle < ((6 * PI / 8) + offset)) { chart = " /     "; }
+                if (abs_angle >= (6 * PI / 8) + offset && abs_angle < ((7 * PI / 8) + offset)) { chart = "-/     "; }
+            }
+
+            return $" {chart} [ {angle,8:F5} rad ]";
+        }
+
+        /// <summary>
+        /// Returns a string for the amplitude's polar representation (magnitude/angle).
+        /// </summary>
+        private string FormatPolar(double magnitude, double angle) =>
+            $"{FormatMagnitude(magnitude, angle)}{FormatAngle(magnitude, angle)}";
+
+        /// <summary>
+        /// Returns a string for the amplitude's cartesian representation (real + imagnary).
+        /// </summary>
+        private string FormatCartesian(double real, double img) =>
+            $"{real,9:F6} + {img,9:F6} i";
+
+        /// <summary>
+        /// The method to use to format the amplitude into a string.
+        /// </summary>
+        private string Format(string label, double real, double img)
+        {
+            var amplitude = (real * real) + (img * img);
+            var angle = System.Math.Atan2(img, real);
+
+            return $"{label}:\t" +
+                   $"{FormatCartesian(real, img)}\t == \t" +
+                   $"{FormatPolar(amplitude, angle)}";
+
+        }
+
+    }
+
+
+    /// <summary>
+    /// Samples a random label from the wavefunction, proportional to its amplitude squared,
+    /// and returns it as a boolean array
+    /// </summary>
+    partial class Sample
+    {
+        private SparseSimulator simulator;
+        public class Native : Sample
+        {
+            public Native(IOperationFactory m) : base(m)
+            {
+                simulator = m as SparseSimulator;
+            }
+            public override Func<IQArray<Qubit>, IQArray<bool>> __Body__ => (__in__) =>
+            {
+                if (simulator != null)
+                {
+                    return new QArray<bool>(((SparseSimulatorProcessor)simulator.QuantumProcessor).Sample(__in__));
+                }
+                return new QArray<bool>( new bool[0]);
+            };
+        }
+    }
+
+    public partial class SparseSimulatorProcessor : QuantumProcessorBase {
```

Comment:
It looks like this still uses `QuantumProcessorBase` / `IQuantumProcessor` rather than `SimulatorBase` / `IQSharpCore`?

Unrelated nit, but code style for C# should be with the brace on the next line.